linalg/sparsemat.cpp

```diff
@@ -3341,6 +3379,10 @@ void SparseMatrix::PrintCSR(std::ostream & os) const
 
    os << height << '\n';  // number of rows
 
+   // HostRead forces synchronization
+   HostReadI();
+   HostReadJ();
+   HostReadData();
```

Comment:
Isn't this misleading? If you print a CSR matrix you want the state of the data represented in a string. What you do here is a side effect and falsely claims that you did the right thing before.

Reply:
The actual state used for computation is at the device, so you want to update the host before printing it and not to print an outdated copy. Moreover, I have copied this from `Print`, so it is not a new behavior.

Reply:
I just added notes to doxygen to let users know 😉 .